tools/push-images

```diff
@@ -31,7 +31,24 @@ export KO_DOCKER_REPO="${IMAGE_REPO}"
 if [[ -z "${IMAGE_TAG:-}" ]]; then
   IMAGE_TAG=$(git rev-parse --short HEAD)-$(date +%Y%m%dT%H%M%S)
 fi
-echo "IMAGE_TAG=${IMAGE_TAG}"
 
-go run github.com/google/ko@v0.14.1 build --tags=${IMAGE_TAG} --base-import-paths --push=true ./cmd/cloud-controller-manager/
-go run github.com/google/ko@v0.14.1 build --tags=${IMAGE_TAG} --base-import-paths --push=true ./cmd/gcp-controller-manager/
+if [[ "$IMAGE_TAG" == *"/"* ]]; then
+  COMPONENT=$(dirname "${IMAGE_TAG}")
+  IMAGE_TAG=$(basename "${IMAGE_TAG}")
+  echo "COMPONENT=${COMPONENT}, IMAGE_TAG=${IMAGE_TAG}"
+else
+  COMPONENT=""
+  echo "IMAGE_TAG=${IMAGE_TAG}"
+fi
+
+if [[ "${COMPONENT:-ccm}" == "ccm" ]]; then
+  go run github.com/google/ko@v0.14.1 build --tags=${IMAGE_TAG} --base-import-paths --push=true ./cmd/cloud-controller-manager/
+else
+  echo "Skipping CCM build, because component is ${COMPONENT}"
+fi
+
+if [[ "${COMPONENT:-gcm}" == "gcm" ]]; then
```

Comment:
Do we have a "short-name" for the `gcp-controller-manager` component?

Reply:
I see one tag as `auth-provider-gcp` https://github.com/kubernetes/cloud-provider-gcp/releases/tag/auth-provider-gcp%2Fv0.0.2

@jprzychodzen @sdmodi  do you know who can help us here?

Reply:
AFAIR `gcp-controller-manager` follows only master branch and there are no published images.

@vinayakankugoyal could you confirm?

Reply:
gcp-controller-manager images are private as they run on the KCP VM.

Reply:
@vinayakankugoyal this is different, this code is OSS and we can perfectly publish images so any external OSS project like kops, or clusterapi, ... can create integration of kubernetes clusters that run in GCE

Reply:
clarified in https://github.com/kubernetes/cloud-provider-gcp/pull/614#discussion_r1344867350, I was confused